getting-started-with-directus-and-flask/index.md

```diff
@@ -0,0 +1,396 @@
+
+## Before You Start
+
+You will need:
+
+- To have Python installed on your machine
+- A Directus project - [follow our quickstart guide](https://docs.directus.io/getting-started/quickstart) if you don't already have one.
+- Basic knowledge of Python and Flask
+
+## Setting Up A Flask Project
+
+To create a new Flask project using `venv`, create your project directory and enter it, then run the following commands:
+
+```sh
+python3 -m venv .venv
+source .venv/bin/activate  # On Windows you should use `.venv\Scripts\activate`
+pip install Flask requests python-dotenv
+```
+
+To make *.env* file variables available on the project, create a file called *config.py* and add the following content to it:
+
+```py
+from dotenv import load_dotenv
+
+load_dotenv()
+```
+
+Then create an *app.py* file to start the Flask app:
+
+```py
+from flask import Flask
+
+app = Flask(__name__)
+
+
+@app.get("/")
+def home():
+    return "Hello world"
+```
+
+And run the following command to run your flask server, which will start a server at [http://localhost:3000](http://localhost:3000):
+
+```sh
+flask run --debug
+```
+
+:::info
+
+The `--debug` flag enables the autoreload feature on the project, which is very useful while developing.
+
+:::
+
+## Creating Global Metadata And Settings Collection
+
+In your Directus project, navigate to **Settings** -> **Data Model** and create a new collection called `global`. Under the **Singleton** option, select 'Treat as a single object', as this collection will have just a single entry containing global website metadata.
+
+Create two text input fields - one with the key of `title` and one `description`.
+
+Navigate to the content module and enter the `global` collection. Collections will generally display a list of items, but as a singleton, it will launch directly into the one-item form. Enter information in the title and description field and hit save.
+
+![Global metadata fields filled with custom text](global-collection-content.png)
+
+By default, new collections are not accessible to the public. Navigate to **Settings** -> **Access Control** -> **Public** and give Read access to the Global collection.
+
+## Creating a Directus Module
+
+Since your data will be fetched via the Directus REST API, you will need to create a module that encapsulates all that logic and exposes a simple interface to the outside world.
+
+To accomplish this, create a *directus.py* file and add the following content to it:
+
+```py
+import requests
+import os
+
+DIRECTUS_BASE_URL = os.environ.get("DIRECTUS_BASE_URL")
+
+
+def get_global_data():
+    response = requests.get(f"{DIRECTUS_BASE_URL}/items/global")
+    return response.json().get("data")
+```
+
+By now this is all you need, but in the following sections, you will also create a new function to fetch data from other collections.
```

Comment:
```suggestion
Since your data will be fetched via the Directus REST API, you will need to create a module that encapsulates all that logic and exposes an interface to the outside world.

To accomplish this, create a `directus.py` file and add the following content to it:

```py
import requests
import os

DIRECTUS_BASE_URL = os.environ.get("DIRECTUS_BASE_URL")


def get_global_data():
    response = requests.get(f"{DIRECTUS_BASE_URL}/items/global")
    return response.json().get("data")
```

By now this is all you need, but in the following sections, you will also create a new function to fetch data from other collections.
```

https://justsimply.dev/